openapi_python_client/parser/openapi.py

```diff
@@ -292,6 +299,21 @@ def _add_parameters(
 
         return endpoint, schemas
 
+    @staticmethod
+    def sort_parameters(*, endpoint: "Endpoint") -> Union["Endpoint", ParseError]:
+        endpoint = deepcopy(endpoint)
+        parameters_from_path = re.findall(_PATH_PARAM_REGEX, endpoint.path)
+        try:
+            endpoint.path_parameters.sort(key=lambda p: parameters_from_path.index(p.name))
+        except ValueError:
+            pass  # We're going to catch the difference down below
+        path_parameter_names = [p.name for p in endpoint.path_parameters]
+        if parameters_from_path != path_parameter_names:
```

Comment:
Does this work if the URL also has string constants in the middle?

e.g. `buildings/<building_id>/floors/<floor_id>/rooms/<room_id>`

Reply:
Yup, and I just added a test to ensure it.